tools/loader/Loader.cpp

```diff
@@ -116,17 +116,85 @@ llvm::cl::opt<std::string>
     emitBundle("emit-bundle",
                llvm::cl::desc("Output directory for the bundle serialization"),
                llvm::cl::cat(loaderCat));
+
+/// Name of the network being bundled.
+llvm::cl::opt<std::string> networkName(
+    "network-name",
+    llvm::cl::desc("Name of the network being bundled. "
+                   "This name is used as both the function name "
+                   "of the entry point to the network "
+                   "and as a prefix for all the files that are generated."),
+    llvm::cl::cat(loaderCat));
 } // namespace
 
 bool glow::emittingBundle() { return !emitBundle.empty(); }
 
+/// Given a string representing a \p path of the form
+/// "path/to/dir" returns the string representing the last
+/// directory (in this case "dir").
+/// \note This is similar to the basename(1) command line tool.
+llvm::StringRef getLastDirectoryName(llvm::StringRef path) {
```

Comment:
@qcolombet LLVM provides convenience APIs for working with file systems. See e.g. `lib/Support/Path.h`. Eventually this whole function can be replaced by a couple of calls of those APIs.

Reply:
Oh thanks!
I didn't know LLVM had that!!

Reply:
Much better with the LLVM implementation, though you have to read the fine print :).
In particular, this part:
/// Path iterator.
[...]
/// * Dot, if one or more trailing non-root slash characters are present.